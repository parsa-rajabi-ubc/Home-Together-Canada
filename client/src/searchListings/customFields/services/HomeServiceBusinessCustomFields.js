/**
 * @Author:     Alex Qin
 * @Created:    2021.02.17
 *
 * @Description: Re-useable HomeServiceBusiness custom fields.
 *
 */
import PropTypes from "prop-types";
import React from 'react';
import HTC_Logo from "../../../images/HTC_Logo.jpg";

const HOME_SERVICE_BUSINESS_CUSTOM_FIELDS_TEXT = {
    rateAndFees: "Rates and fees ($CAD)",
    pictures: "Pictures (max of 6)"
}


const HomeServiceBusinessCustomFields = (props) =>{
    const { title, rateAndFees, pictures } = props;

    return(
        <div>
            <h1> {title} </h1> <br/>
            <p>{HOME_SERVICE_BUSINESS_CUSTOM_FIELDS_TEXT.rateAndFees} {rateAndFees}</p> <br/>
            <p>{HOME_SERVICE_BUSINESS_CUSTOM_FIELDS_TEXT.pictures}</p> <br/>
            {/*TODO: replace HTC_Logo with pictures Pictures (max of 6)*/}
            <img className={"float-left w-24 h-auto mx-4 "} src={HTC_Logo} alt={""}/> <br/>
        </div>
    );
}

HomeServiceBusinessCustomFields.propTypes = {
    title: PropTypes.string.isRequired,
    rateAndFees: PropTypes.number.isRequired,
    pictures: PropTypes.string.isRequired
}

export default HomeServiceBusinessCustomFields;