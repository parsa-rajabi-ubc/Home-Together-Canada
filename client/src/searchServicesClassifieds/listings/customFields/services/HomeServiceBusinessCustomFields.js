/**
 * @Author:     Alex Qin
 * @Created:    2021.02.17
 *
 * @Description: Re-usable HomeServiceBusiness custom fields.
 *
 */
import PropTypes from "prop-types";
import React from 'react';
import {ImageGallery} from "../../../../common/listings/ImageGallery";

const HOME_SERVICE_BUSINESS_CUSTOM_FIELDS_TEXT = {
    RATE_AND_FEES: "Rates and fees (CAD)",
    PICTURES: "Pictures"
}


const HomeServiceBusinessCustomFields = (props) =>{
    const { rateAndFees, pictures } = props;
    console.log('HomeServiceBusinessCustomFields props: ', props);

    return(
        <div>
            <label className={"label-result"}>{HOME_SERVICE_BUSINESS_CUSTOM_FIELDS_TEXT.RATE_AND_FEES}</label>
            <p> {rateAndFees}</p>
            {!!pictures.length &&
                <div>
                    <label className={"label-result"}>{HOME_SERVICE_BUSINESS_CUSTOM_FIELDS_TEXT.PICTURES}</label>
                    <ImageGallery pictures={pictures}/>
                </div>
            }
        </div>
    );
}

HomeServiceBusinessCustomFields.propTypes = {
    rateAndFees: PropTypes.string.isRequired,
    pictures: PropTypes.string
}

export default HomeServiceBusinessCustomFields;