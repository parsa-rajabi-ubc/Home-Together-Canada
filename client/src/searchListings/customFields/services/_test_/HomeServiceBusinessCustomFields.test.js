/**
 * @Author:     Alex Qin
 * @Created:    2020.01.14
 *
 * @Description: HomeServiceBusinessCustomFields snapshot test. Returns a snapshot.
 *
 */
import React from 'react';
import renderer from  'react-test-renderer'
import {BrowserRouter as Router} from "react-router-dom";
import {Facilitation} from "../MockServices";
import HomeServiceBusinessCustomFields from "../HomeServiceBusinessCustomFields";

jest.mock("react-tooltip/node_modules/uuid", () => ({
            v4: () => "00000000-0000-0000-0000-000000000000"}
    )
);

jest.mock('react-redux', () => ({
    connect: () => {
        return (component) => {
            return component
        };
    }
}));

describe('GovernmentServicesCustomFields', () => {
    it("should render correctly regardless of properties", () => {
        // given
        const reset = jest.fn();
        const props = {
            ...Facilitation,
            reset
        };
        //when
        const component = renderer.create(<Router><HomeServiceBusinessCustomFields {...props}/></Router>);
        const tree = component.toJSON();
        //then
        expect(tree).toMatchSnapshot();
    });
});