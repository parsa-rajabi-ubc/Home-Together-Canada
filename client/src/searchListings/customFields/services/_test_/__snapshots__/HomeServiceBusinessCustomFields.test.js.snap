// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovernmentServicesCustomFields should render correctly regardless of properties 1`] = `
<div>
  <h1>
     
    Facilitation Mock data
     
  </h1>
  <p>
    Rates and fees ($CAD)
     
    666
  </p>
  <p>
    Pictures (max of 6)
  </p>
  <img
    alt=""
    src="HTC_Logo.jpg"
  />
</div>
`;
